File: branches/Andean/Fantasy.Base/ServiceModel/TaskBarProgressMonitorService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Fantasy.ServiceModel
{
    public class TaskBarProgressMonitorService : TaskBarProgressMonitor, IService, IObjectWithSite
    {

        private IProgressMonitorContainer _container;
        public virtual void InitializeService()
        {
            if(this.Site != null)
            {
                _container = this.Site.GetService<IProgressMonitorContainer>();
                if (_container != null)
                {
                    _container.AddMoniter(this);
                }
            }

            

            if (this.Initialize != null)
            {
                this.Initialize(this, EventArgs.Empty);
            }
        }

        public virtual void UninitializeService()
        {
            if (this.Uninitialize != null)
            {
                this.Uninitialize(this, EventArgs.Empty);
            }
            if (_container != null)
            {
                this._container.RemoveMoniter(this);
            }
        }

        public event EventHandler Initialize;

        public event EventHandler Uninitialize;

    
        public IServiceProvider Site { get; set; }
      
    }
}

File: branches/Stheno/AutoConverted/Fantasy.Base/ServiceModel/TaskBarProgressMonitorService.java
package Fantasy.ServiceModel;

public class TaskBarProgressMonitorService extends TaskBarProgressMonitor implements IService, IObjectWithSite
{

	private IProgressMonitorContainer _container;
	public void InitializeService()
	{
		if(this.getSite() != null)
		{
			_container = this.getSite().<IProgressMonitorContainer>GetService();
			if (_container != null)
			{
				_container.AddMoniter(this);
			}
		}



		if (this.Initialize != null)
		{
			this.Initialize(this, EventArgs.Empty);
		}
	}

	public void UninitializeService()
	{
		if (this.Uninitialize != null)
		{
			this.Uninitialize(this, EventArgs.Empty);
		}
		if (_container != null)
		{
			this._container.RemoveMoniter(this);
		}
	}

//C# TO JAVA CONVERTER TODO TASK: Events are not available in Java:
//	public event EventHandler Initialize;

//C# TO JAVA CONVERTER TODO TASK: Events are not available in Java:
//	public event EventHandler Uninitialize;


	private IServiceProvider privateSite;
	public final IServiceProvider getSite()
	{
		return privateSite;
	}
	public final void setSite(IServiceProvider value)
	{
		privateSite = value;
	}

}